interfaces/builtin/gpio_chardev_test.go

```diff
@@ -0,0 +1,277 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/check.v1"
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/features"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/interfaces/kmod"
+	"github.com/snapcore/snapd/interfaces/systemd"
+	"github.com/snapcore/snapd/interfaces/udev"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type GpioChardevInterfaceSuite struct {
+	testutil.BaseTest
+
+	iface    interfaces.Interface
+	slot     *interfaces.ConnectedSlot
+	slotInfo *snap.SlotInfo
+	plug     *interfaces.ConnectedPlug
+	plugInfo *snap.PlugInfo
+}
+
+var _ = Suite(&GpioChardevInterfaceSuite{
+	iface: builtin.MustInterface("gpio-chardev"),
+})
+
+const gpioChardevGadgetYaml = `name: my-device
+version: 0
+type: gadget
+slots:
+  gpio-chardev-good:
+    interface: gpio-chardev
+    source-chip:
+      - chip0
+      - chip1
+    lines: 3,4,1-2,5
+  no-source-chip-attr:
+    interface: gpio-chardev
+    lines: 3,4,1-2,5
+  no-lines-attr:
+    interface: gpio-chardev
+    source-chip: [chip2]
+  duplicate-source-chip:
+    interface: gpio-chardev
+    source-chip: ["chip","chip"]
+    lines: 3,4,1-2,5
+  duplicate-line:
+    interface: gpio-chardev
+    source-chip: [chip3]
+    lines: 2-6,3
+  bad-source-chip-0:
+    interface: gpio-chardev
+    source-chip: []
+    lines: 3,4,1-2,5
+  bad-source-chip-1:
+    interface: gpio-chardev
+    source-chip: [" s"]
+    lines: 3,4,1-2,5
+  bad-source-chip-2:
+    interface: gpio-chardev
+    source-chip: ["s "]
+    lines: 3,4,1-2,5
+  bad-source-chip-3:
+    interface: gpio-chardev
+    source-chip: [""]
+    lines: 3,4,1-2,5
+  bad-range-0:
+    interface: gpio-chardev
+    source-chip: [chip4]
+    lines: 2-
+  bad-range-1:
+    interface: gpio-chardev
+    source-chip: [chip5]
+    lines: a-3
+  bad-range-2:
+    interface: gpio-chardev
+    source-chip: [chip6]
+    lines: 0-10000000
+  bad-range-3:
+    interface: gpio-chardev
+    source-chip: [chip7]
+    lines: 4-2
+  bad-range-4:
+    interface: gpio-chardev
+    source-chip: [chip8]
+    lines: 0--1
+  bad-line-0:
+    interface: gpio-chardev
+    source-chip: [chip9]
+    lines: a
+  bad-line-1:
+    interface: gpio-chardev
+    source-chip: [chip10]
+    lines: "-1"
+  bad-lines-count:
+    interface: gpio-chardev
+    source-chip: [chip11]
+    lines: 0,1-65536
+`
+
+const gpioChardevConsumerYaml = `name: consumer
+version: 0
+apps:
+  app:
+    plugs:
+      - gpio-chardev-good
+plugs:
+  gpio-chardev-good:
+    interface: gpio-chardev
+`
+
+func (s *GpioChardevInterfaceSuite) SetUpTest(c *C) {
+	dirs.SetRootDir(c.MkDir())
```

Comment:
the test should mock os release, otherwise the libexecdir paths won't match if executing on opensuse or fedora